test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscovererTests.cs

```diff
@@ -247,6 +248,174 @@ public async Task DiscoverAsync_IsIdempotent()
             }
         }
 
+        [PlatformTheory(Platform.Windows)]
+        [InlineData("nuget-plugin-myPlugin.exe")]
+        [InlineData("nuget-plugin-myPlugin.bat")]
+        public async Task DiscoverAsync_withValidDotNetToolsPluginWindows_FindsThePlugin(string fileName)
+        {
+            using (var testDirectory = TestDirectory.Create())
+            {
+                // Arrange
+                var pluginPath = Path.Combine(testDirectory.Path, "myPlugin");
+                Directory.CreateDirectory(pluginPath);
+                var myPlugin = Path.Combine(pluginPath, fileName);
+
+                Environment.SetEnvironmentVariable("PATH", pluginPath);
```

Comment:
Environment variables are per-process, which means that `Environment.SetEnvironmentVariable` is changing it for the entire process. If there are other tests running in parallel that depend on other values, this will cause those tests to become flakey.

The general pattern we use thoughout NuGet's source code is to use an interface named something like `IEnironmentVariableHelper` in the product code, so that in tests we can pass in a mock. And in the product code where we need to use real environment variables, there's something like `EnvironmentVariableHelper.Instance`.

Basically, tests should never modify shared state.